measurements/api/measurements.py

```diff
@@ -214,21 +269,84 @@ def input_cte(input_, domain, test_name):
             text('input.input LIKE :i').bindparams(i='%{}%'.format(input_)),
         )
 
-    if domain:
-        domain_filter = '{}%'.format(domain)
-        where_or.append(text('input.input LIKE :domain_filter').bindparams(domain_filter=domain_filter))
-        if test_name in [None, 'web_connectivity', 'http_requests']:
-            where_or.append(text('input.input LIKE :http_filter').bindparams(http_filter='http://{}'.format(domain_filter)))
-            where_or.append(text('input.input LIKE :https_filter').bindparams(https_filter='https://{}'.format(domain_filter)))
-
-    url_q = select([
-        column("input").label('input'),
-        column("input_no").label('input_no'),
-    ]).where(
-        or_(*where_or)
-    ).select_from(sql.table('input'))
+    else:
+        domain_filter = "{}%".format(domain)
+        where_or.append(
+            text("input.input LIKE :domain_filter").bindparams(
+                domain_filter=domain_filter
+            )
+        )
+        if test_name in [None, "web_connectivity", "http_requests"]:
+            where_or.append(
+                text("input.input LIKE :http_filter").bindparams(
+                    http_filter="http://{}".format(domain_filter)
+                )
+            )
+            where_or.append(
+                text("input.input LIKE :https_filter").bindparams(
+                    https_filter="https://{}".format(domain_filter)
+                )
+            )
+
+    url_q = (
+        select([column("input").label("input"), column("input_no").label("input_no")])
+        .where(or_(*where_or))
+        .select_from(sql.table("input"))
+    )
+
     return url_q.cte('input_cte')
 
+
+def log_query(log, q):
+    import sqlparse  # debdeps: python3-sqlparse
+    sql = str(q.statement.compile(dialect=postgresql.dialect(),
+            compile_kwargs={"literal_binds": True}))
+    sql = sqlparse.format(sql, reindent=True, keyword_case='upper')
+    log.info("\n--- query ---\n\n%s\n\n-------------", sql)
+
+
+def _merge_two_results(a, b):
+    """Merge 2 measurements. Collect useful fields from traditional pipeline
+    and fastpath
+    """
+    if a["scores"]:
+        # a is fastpath
+        if b["scores"]:
+            # both fastpath, ignore b
+            return a
+
+        else:
+            # merge in useful fields from traditional
+            for f in ("anomaly", "confirmed"):
+                a[f] = b[f]
+            return a
+
+    else:
+        if b["scores"]:
+            # merge in useful fields from fastpath
+            for f in ("scores", "measurement_url", "measurement_id"):
+                a[f] = b[f]
+            return a
+
+        else:
+            # both trad, ignore b
+            return a
```

Comment:
Maybe we can improve a bit the readability of this function by reducing the level of nesting and explcitly listing all the cases, for example:
```
if a["scores"] and b["scores"]:
  # both fastpath, ignore b
  return a
elif a["scores"]:
  # merge in useful fields from traditional
  ...
elif b["scores"]:
  # merge in useful fields from fastpath
 ...
else:
  # both trad, ignore b
 ...      
```

What do you think?

Reply:
Done